forge-game/src/main/java/forge/game/ability/effects/ConniveEffect.java

```diff
@@ -68,7 +68,7 @@ public void resolve(SpellAbility sa) {
             CardCollection connivers = CardLists.filterControlledBy(toConnive, p);
             while (!connivers.isEmpty()) {
                 GameEntityCounterTable table = new GameEntityCounterTable();
-                final CardZoneTable triggerList = new CardZoneTable(sa.getLastStateBattlefield(), sa.getLastStateGraveyard());
+                final CardZoneTable triggerList = new CardZoneTable(game.getLastStateBattlefield(), game.getLastStateGraveyard());
```

Comment:
The SpellAbility one is only really filled when
a) it's the castSA for stuff like "as you cast" (Presence)
b) we're inside a replacement ability (which can then be used inside AbilityUtils valid checks)

so most of the time it's unavailable for normal LKI usage inside the table and we should move away from using it in effects? 🤔 
And I suppose there can be scenarios where we need to use `game.copyLastState` because the previous one might be stale?